### ISSUE TITLE
After a few minutes, kafkacat instances cannot be killed with SIGTERM or SIGINT

### ISSUE THREAD

**Issue by JamesDunne** [edenhill/kafkacat#308]
On MacOS 10.14 Mojave, after I leave a `kafkacat` instance running in consumer mode for more than a few minutes, I'm unable to CTRL-C it or `kill <pid>` it. I have to resort to a `kill -9 <pid>` to actually kill the instance.

I started the instance in the `-d all` state and see a constant stream of debug output as it's successfully polling from an idle topic. Every time I attempt a CTRL-C it's like it's completely ignored by the tool. There's no debug output to indicate it even caught the signal or handled it.

I should note I'm connecting to a secured kafka cluster with 4 brokers using SASL_SSL.

Using kafkacat 1.6.0 installed via homebrew on MacOS:
```
$ kafkacat -V
kafkacat - Apache Kafka producer and consumer tool
https://github.com/edenhill/kafkacat
Copyright (c) 2014-2019, Magnus Edenhill
Version 1.6.0 (JSON, Avro, Transactions, librdkafka 1.5.2 builtin.features=gzip,snappy,ssl,sasl,regex,lz4,sasl_gssapi,sasl_plain,sasl_scram,plugins,zstd,sasl_oauthbearer)
```

**Comment by edenhill** [comment]
Interesting, can you provide your full command line?

**Comment by JamesDunne** [comment]
It was very basic, nothing special going on. Just `kafkacat -C -t topic -J -d all` plus a bunch of sasl options pulled from a conf file.

**Comment by edenhill** [comment]
This is most likely a hang in consumer_close(), there's been fixes for that in librdkafka since 1.5.2. Suggest you upgrade

**Comment by JamesDunne** [comment]
Looks like it's resolved in librdkafka 1.6.1 although it takes a few seconds to respond to the CTRL-C. Thanks!
